designs/2022-community-eslint-org/README.md

```diff
@@ -0,0 +1,263 @@
+would all be admin of the org.
+
+We can make a team (with sub-teams if we want) for each package, where we give
+admin rights on the repo for that specific package.
+
+The core team would help maintain all repos, whereas people in specific teams
+would only help maintain these repositories.
+
+**Bonus:** I think having our own place in the Discord server where we can
+discuss some things would be a good idea as well.  
+This way the "core" team has an open place to discuss some common things like
+extracting common functionality into a separate package, minimum Node & ESLint
+target, how to handle disputes between maintainers, ...  
+Having a channel for each separate package would maybe be a good idea as well,
+so the community has a change to ask questions to the maintainers as well.
+
+### What should be the role of the ESLint team?
+
+I think most important role would be to facilitate at least the contact with
+`@mysticatea`, so we can transfer his packages over to the `@eslint-community`
+org & gain `npm` publish rights for the packages.
+
+Next to that, I can see the ESLint team talk to the "core" team of the new org
+regarding creating new community maintained packages that are split out from the
+main ESLint repo. Examples of these kind of repos are:
+
+- `eslint-plugin-node`  
+  This plugin was recommended because of the deprecation of some rules in v7  
+  <https://eslint.org/docs/user-guide/migrating-to-7.0.0#deprecate-node-rules>
+- `eslint-formatter-codeframe` & `eslint-formatter-table`  
+  These were both in the main ESLint repo but were removed in v8 and are
+  currently maintained by [@fregante](https://github.com/fregante)  
+  <https://eslint.org/docs/8.0.0/user-guide/migrating-to-8.0.0#-removed-codeframe-and-table-formatters>
+- `eslint-plugin-eslint-plugin`  
+  This plugin is (together with `eslint-plugin-node`) a recommended plugin for
+  linting custom plugins.  
+  <https://eslint.org/docs/developer-guide/working-with-plugins#linting>
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+I don't think this RFC needs a formal announcement on the ESLint blog.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think
+    about any opposing viewpoints that reviewers might bring up.
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+I can't think of a reason why we would not want this.  
+Once we have access to the `@eslint-community` org & have the repos transfered,
+the "core" team can work out the rest of the process themselves.
```

Comment:
The drawback is that this may puncture the wider community and give an impression of the modules being blessed / recommended by the core team and thus making other initiatives harder.

It could also put pressure on modules from other ESLint related organizations / maintainers to transfer their projects here as this is the "correct" place for popular packages.

Overall that can create quite a large organization where lots of different minds are forced together when then actually would thrive better when being organized in a more pluralistic landscape of multiple organizations.

Some actual considerations:

- How does this relate to stuff in eg. https://github.com/standard/ – would there be an expectency from the wider community or the eslint-community project itself to transfer some stuff from there?
- How about projects like https://github.com/sindresorhus/eslint-plugin-unicorn – will it be considered a problem that such a plugin is under @sindresorhus ownership rather than the eslint-community?
- How about something like [`eslint-config-airbnb-base`](https://www.npmjs.com/package/eslint-config-airbnb-base)? Could a ruleset like that be eligible for this community if abandoned at some point? Would there be a push by people to transfer such a module here? Will that make it look like an officially endorsed configuration? How would that affect projects like StandardJS?

Reply:
good question! I don't think we want to accept share configs. If a shared config is no longer maintained, the users can:
* maintain a fork
* switch to another